### PR TITLE
fix(pagination): fix pagination with explicit start page

### PR DESCRIPTION
**Pull request by maheshwariST**

### CODE REVIEW

File: Octokit/Http/ApiConnection.cs [hunk 1]
```diff
@@ -637,7 +643,7 @@ public async Task<IReadOnlyList<T>> GetQueuedOperation<T>(Uri uri, CancellationT
                         options);
 
                     return shouldContinue
-                        ? connection.Get<List<TU>>(nextPageUri, parameters, accepts)
+                        ? connection.Get<List<TU>>(nextPageUri, parametersWithoutStartPage, accepts)
```
Comment (maheshwariST) [inline Octokit/Http/ApiConnection.cs:646]: This fix is unrelated to my previous changes. If this fix were to be not applied then this block overwrites the page parameter in the `nextPageUrl` set by the `ReadOnlyPagedCollection.GetNextPage` method. And this becomes endless loop.

**Review by afilatov-st — APPROVED** [review]
